docs/configuration.md

```diff
@@ -20,6 +20,23 @@ The following table shows the available config keys for Deptrac.
 </thead>
 <tbody>
 <tr>
+<td>analyser.internal_tag</td>
+<td>
+Specifies which doc block tag deptrac should use to identify layer-internal class-like tokens.
+The default is <code>"@internal"</code>. May be set to <code>null</code> to disable.
```

Comment:
https://github.com/qossmic/deptrac/issues/1287 as mentioned in the issue i like to remove `@internal` as the layer-internal tag. Therefor would you instead make `deptrac-internal` the default?
https://github.com/qossmic/deptrac/blob/main/docs/upgrade.md should also be updated! 
In case anyone wants to use `@internal` than it's possible to just change the setting.

Reply:
So, you are proposing to make @deptrac-internal the default, instead of always supporting it?

That would be a breaking change for users depending on the fact that @internal works out of the box... They also wouldn't necessarily notice, since the new behavior would simply ignore defects that were previously detected.

I like it since it's cleaner, but is this an acceptable change in behavior from a compatibility point of view? Who can make that call?

Reply:
Right now I don't see why we should support multiple annotations for this purpose?
But I'm open to support multiple annotations if it makes sense.

Deptrac misused **@internal** for the layer internal and therefore I would like to remove it as the current default.

And because the main branch already contains BCs, the coming release will be v2.

It would have been better to have a transition period, but for me it's acceptable if we provide a fallback and document it.

Or we do as you suggest but add a depreciation notice, that says something like: _the future default will be @deptrac-internal, if you want to keep the current behavior, please set analyzer.internal_tag=@internal_.

@patrickkusebauch what do you think?

Reply:
@brightbyte I'm the current maintainer and right now I'm planing the next big release, which will properly also change the way we distribute Deptrac.

Reply:
> Right now I don't see why we should support multiple annotations for this purpose? But I'm open to support multiple annotations if it makes sense.

I don't either, but that's the status quo. So I didn't want to change more than necessary.

> It would have been better to have a transition period, but for me it's acceptable if we provide a fallback and document it.

If you think it's fine, I'm good :)

> Or we do as you suggest but add a depreciation notice, that says something like: _the future default will be @deptrac-internal, if you want to keep the current behavior, please set analyzer.internal_tag=@internal_.
> 
> @patrickkusebauch what do you think?

A quick code search on github shows no usage of @deptrac-internal anywhere, so we could still go for something different, like layer-internal. Or even default to null.

Reply:
using `@deptrac-*` is the industry standard for annotation/doc tags, see `@phpstan-*` or `@psalm-*`. That's something we should not get rid of. If using `@internal` is contentious, that's what we should drop.

Reply:
So, @deptrac-internal always, plus optionally another one, letting the internal_tag option default to null? Or configurable with @deptrac-internal as the default value of internal_tag?

Reply:
Either:
 - `@deptrac-internal` always, plus optionally another one, letting the `internal_tag` option default to `null` (This is BC change that requires major release)
 - `@deptrac-internal` always, plus optionally another one, letting the `internal_tag` option default to `@internal` (This is not BC change and can be released in minor release)

Reply:
In that case, you should add a section about this into https://github.com/qossmic/deptrac/blob/main/docs/upgrade.md

Reply:
Or at least add write more notes about the BC change into the PR description so that when new release is cut, it is easier for the person doing the job to identify what the BC break of this PR is.